### PR TITLE
Make backend and conftest into links in dev docs

### PR DESCRIPTION
**Pull request by nvaytet**

### CODE REVIEW

File: docs/developer/testing.rst [hunk 1]
```diff
@@ -21,12 +21,18 @@ In addition, the fakes are relatively complex and may diverge from the real impl
 
 To mitigate these problems, the fake client is tested alongside a real client.
 But to (mostly) avoid the downsides stated in the beginning, the real client is connected to a local SciCat server.
-See ``src/testing/backend.py`` and ``tests/conftest.py`` for the concrete setup.
+See the |backend folder|_ folder and |conftest file|_ for the concrete setup.
```
Comment (jl-wynen) [inline docs/developer/testing.rst:24]: Why is this needed? Can't you use 
```
`src/scitacean/backend <https://github.com/SciCatProject/scitacean/tree/main/src/scitacean/testing/backend>`_
```
?

File: docs/developer/testing.rst [hunk 2]
```diff
@@ -21,9 +21,9 @@ In addition, the fakes are relatively complex and may diverge from the real impl
 
 To mitigate these problems, the fake client is tested alongside a real client.
 But to (mostly) avoid the downsides stated in the beginning, the real client is connected to a local SciCat server.
-See ``src/testing/backend.py`` and ``tests/conftest.py`` for the concrete setup.
+See the `src/scitacean/backend <https://github.com/SciCatProject/scitacean/tree/main/src/scitacean/testing/backend>`_ folder and `tests/conftest.py <https://github.com/SciCatProject/scitacean/blob/main/tests/conftest.py>`_ file for the concrete setup.
 The backend is launched in a docker container with the image of the latest release of the SciCat backend.
-Tests in ``tests/client`` are then run with both the fake and the real client to ensure that both produce the same results.
+Tests in `tests/client <https://github.com/SciCatProject/scitacean/tree/main/tests/client>`_ are then run with both the fake and the real client to ensure that both produce the same results.
 
 Use ``--backend-tests`` with ``pytest`` to run these tests.
```
Comment (nvaytet) [inline docs/developer/testing.rst:29]: It seems the paths to the files below are out of date. There is no `tests/common/sftp_server.py` in the repo I think?

Comment (jl-wynen) [reply docs/developer/testing.rst:29]: True. Those files are now packaged with Scitacean and are now in 
```
:mod:`scitacean.testing.sftp`
```

Can you update the paths to this? I don't think we need to refer to any concrete files in this case but to the user guide in https://scicatproject.github.io/scitacean/user-guide/testing.html#Local-SFTP-fileserver

Comment (jl-wynen) [reply docs/developer/testing.rst:29]: Yes: https://scicatproject.github.io/scitacean/generated/modules/scitacean.testing.sftp.fixtures.sftp_fileserver.html#scitacean.testing.sftp.fixtures.sftp_fileserver

**Review by jl-wynen — APPROVED** [review]
Thanks!
